### ISSUE TITLE
[FR] Add "Login page" to Elementor Theme Builder Templates

### ISSUE THREAD

**Issue by VerdiH** [elementor/elementor#8222]
**What problem is your feature request going to solve? Please describe.**

Make it easier and less time consuming for web designers to create a custom login page for their clients.

**Describe the solution you'd like**

Create a custom Login Page using Elementor, replacing the actual login page. Similar to how you create the 404 page using Elementor Pro.

OR.. could be an option in Hello Theme but I rather see that remaining as bare bones as possible.

**Describe alternatives you've considered**

1. Using the Popup Builder to create a new login form (dynamically displaying the username after logging in)>
2. Using the Code Snippets Plugin to replace the default WP Login page, that otherwise stays visible when using only the Popup Builder. See these steps if you want to know the details.

### Steps

1. Create login page using Elementor Pro's Login Widget, note it's slug.
- Create Code Snippet, name it something like "Elementor Login Page"
- Copy-Paste below code in snippet, replace the word 'inloggen' in all four locations with you slug.

```
function redirect_login_page() {
  $login_page  = home_url( '/inloggen/' );
  $page_viewed = basename($_SERVER['REQUEST_URI']);
 
  if( $page_viewed == "wp-login.php" && $_SERVER['REQUEST_METHOD'] == 'GET') {
    wp_redirect($login_page);
    exit;
  }
}
add_action('init','redirect_login_page');

function login_failed() {
  $login_page  = home_url( '/inloggen/' );
  wp_redirect( $login_page . '?login=failed' );
  exit;
}
add_action( 'wp_login_failed', 'login_failed' );
 
function verify_username_password( $user, $username, $password ) {
  $login_page  = home_url( '/inloggen/' );
    if( $username == "" || $password == "" ) {
        wp_redirect( $login_page . "?login=empty" );
        exit;
    }
}
add_filter( 'authenticate', 'verify_username_password', 1, 3);

function logout_page() {
  $login_page  = home_url( '/inloggen/' );
  wp_redirect( $login_page . "?login=false" );
  exit;
}
add_action('wp_logout','logout_page');
```

### Note

- wp-login and wp-admin are now redirected to the login page you created with Elementor. Even after logging out.
- When an error occurs, the user will be redirected back to the login page you created. No error message is given, although Hongkiat has a snippet for that, see title "Error Message". Sadly it need altering the login page template you do not have when making it with Elementor.

---

Please let me know if you need me to set up examples, mockups or if you have any questions.

**Comment by lucafagherazzi** [comment]
+1

**Comment by codeorlov** [comment]
+1

**Comment by ouranos2** [comment]
+1 - That would be very useful and convenient.

**Comment by RichardUUU** [comment]
Yes, please. I'm using a hodgepodge of plugins to get people registered and logged in. Would love to move everything to Elementor.

**Comment by elasticeye** [comment]
+1 for this!

**Comment by rodolphebertozzo** [comment]
+1

**Comment by 79ho3ein** [comment]
+1
